merlin/table/tensor_table.py

```diff
@@ -31,6 +32,10 @@ class TensorTable:
     A DataFrameLike wrapper around a dictionary of arrays or tensors
     """
 
+    @classmethod
+    def from_df(cls, df):
+        return tensor_table_from_df(df)
```

Comment:
In the discussion on the PR that introduced TensorTable there was a mention of inline imports to avoid an import of dataframe libs when using this, is that no longer required here?

Reply:
I still think it's a good idea to decouple this from dataframe libraries as much as possible, but my thinking on how to do that has shifted a bit. Since Pandas and cuDF require NumPy and CuPy (respectively), I think we can convert these functions to use `lazy_singledispatch` (because the availability of the dataframe types also implies availability of the relevant array library.)

I'll take a stab at that and add it to this PR.